### DISCUSSION TITLE
Thermostat card with a different source for temperature

### DISCUSSION THREAD

**Discussion by RobertoBiundo** [home-assistant/frontend#19276] [Dashboard]
###  The issue

One of the main issues with the smart thermostats that sit next to the radiator is that they measure the temperature right next to the radiator. This causes the temperature measurements to be completly off. As an example in a room where the temperature is about 18C (measured with a separate sensor) the thermostat temp is reported as 24C.

### Possible solutions

Now, if this behavior was linear it wouldn't be a problem since we could just apply an offset to it (some of the devices even support this). However that is not the case, as the radiator gets hotter the temperature offset increases and on idle (cold) the temperature offset is 0.

Another solution might be to add some isolation material between the radiator and the thermostat valve, however this is also not a good solution since altough it reduces the offset it still does not fixes the issue.

### A better solution

A better solution is therefore to add new sensor somewhere else in the room (an unexpensive one from SONOFF would serve this purpose perfectly) and measure the temperature somewhere where there are no hear sources. However **the thermostat card does not support the ability to take the temperature from a different entity (the sensor) and control a different one (the thermostat valve)**. This would be a great thing to have.

Since most of these thermostats thave their own PIDs is therefore imposible to do an action like
- Temperature in the thermostat is 28 degrees
- Temperature in the room is actually 18 degrees
- Requested temperature int he room is 20 degrees
- Valve should be open, however the thermostat would still be closed since it has it own PID.
A solution that I have found for this is just to use it as an ON/OFF valve so if we need it open we tell it to go to 50C. If we need it closed we tell it to go to 5C. This works quite well, and most of them have a "boost" funtion that also allows me to override it manually and open or close it at will.

Now, I know that my current solution might not be the best but I can image there are lot of people smarter than me that can figure something more universal that works for everyone. Maybe we need some Helper that converts the thermostaat into an ON/OFF boolean entity, or something else. 

### TL;DR; The request

As a user with an unreliable temperature measurement from my thermostat valve due to it being close to a heat source I would like home assistant to control the valve using it as an ON/OFF valve and taking as current temperature the one from a different entity (a separate temperature sensor). If possible use the "boost" attribute of the thermostaat valve to for the valve open and close when required.
